packages/legend-application-studio/src/components/editor/edit-panel/uml-editor/ClassEditor.tsx

```diff
@@ -193,6 +205,49 @@ const PropertyBasicEditor = observer(
       setUpperBound(event.target.value);
       updateMultiplicity(lowerBound, event.target.value);
     };
+    // Drag and Drop
+    const handleHover = useCallback(
+      (item: ClassPropertyDragSource, monitor: DropTargetMonitor): void => {
+        const dragIndex = editorState.classState.propertyStateTesting.findIndex(
+          (e) => e === item.property,
+        );
+        const hoverIndex =
+          editorState.classState.propertyStateTesting.findIndex(
+            (e) => e === property,
+          );
+
+        if (dragIndex === -1 || hoverIndex === -1 || dragIndex === hoverIndex) {
+          return;
+        }
+        // move the item being hovered on when the dragged item position is beyond the its middle point
+        const hoverBoundingReact = ref.current?.getBoundingClientRect();
+        const distanceThreshold =
+          ((hoverBoundingReact?.bottom ?? 0) - (hoverBoundingReact?.top ?? 0)) /
+          2;
+        const dragDistance =
+          (monitor.getClientOffset() as XYCoord).y -
```

Comment:
```suggestion
          (monitor.getClientOffset()?.y ?? 0) -
```

Also, remove usage of `XYCoord`

Reply:
Cool, I will go ahead to switch to nullish coalescing and removing XYCoord in the query/generation specification editor as well